File: asonicOFDM/app/src/main/java/com/example/asonicofdm/OFDMConfig.java
package com.example.asonicofdm;

public class OFDMConfig {

    public int carrier_count;
    public int symbol_per_carrier;
    public int bits_per_symbol;
    public int IFFT_length;
    public int d;
    public double PrefixRatio;
    public double beta;
    public int Fc;
    public int Fs;
    public int GI;
    public int GIP;
    public int[] carriers;
    public int[] conjugate_carriers;
    public int baseband_length;
    public int modulate = 2;
    public int[] preamble;

    public OFDMConfig(boolean is_pre) {
        if (is_pre) {
            carrier_count = 4;
            symbol_per_carrier =2;
            bits_per_symbol = 2;
            IFFT_length = 256;
        }
        else {
            carrier_count = 16;
            symbol_per_carrier = 16;
            bits_per_symbol = 2;
            IFFT_length = 512;
        }
        d = 5;
        PrefixRatio = (double)1 / 4;
        beta = (double)1 / 32;
        Fc = 17000;
        Fs = 10000;
        GI = (int)(PrefixRatio * IFFT_length);
        GIP = (int)(beta * (IFFT_length + GI));
        carriers = new int[carrier_count];
        conjugate_carriers = new int[carrier_count];
        for (int i = 0; i < carrier_count; i++)
            carriers[i] = i + (IFFT_length / 5) - carrier_count / 2;
        for (int i = 0; i < carrier_count; i++)
            conjugate_carriers[i] = IFFT_length - carriers[i];
        baseband_length = carrier_count * symbol_per_carrier * bits_per_symbol;
        modulate = 2;
        preamble = new int[]{0, 0, 0, 0, 1, 0, 1, 1, 0, 1, 1, 1, 0, 1, 1, 1};
    }
}

File: Matlab_2/ofdm_config.m
function [config] = ofdm_config(preamble)
%
% Syntax: [config], ofdm_config()
%
% Return the default ofdm config info
    if (nargin < 1) preamble = false;
    end
    if (preamble)
        % 使用的子载波 四个
        config.carriers = [31,32,33,34];
        % 每个子载波符号数
        config.symbol_per_carrier = 2;
        % 每个符号比特数
        config.bits_per_symbol = 2;
    else
        % 使用的子载波 所有
        config.carriers = 1:64;
        % 每个子载波符号数
        config.symbol_per_carrier = 16;
        % 每个符号比特数
        config.bits_per_symbol = 2;
    end
    % 子载波数
    config.carrier_count = length(config.carriers);
    % FFT点数
    config.IFFT_length = 64;
    % 每个符号距离
    config.d = 5;
    % 保护间隔与OFDM数据的比例 1/6~1/4
    config.PrefixRatio = 1/4;
    % 窗函数滚降系数
    config.beta = 1/8;
    % 声音信号频率 (载波频率)
    config.Fc = 17000;
    % 声音信号采样率
    config.Fs = 10000;
    % 每一个OFDM符号添加的循环前缀长度为1/4*IFFT_length
    config.GI = config.PrefixRatio * config.IFFT_length;
    % 循环后缀的长度
    config.GIP = config.PrefixRatio * config.IFFT_length;
    % 可携带的信息比特数
    config.load_bits = config.carrier_count * config.symbol_per_carrier * config.bits_per_symbol;
    % 调制方式, 1 -> qam16, 2 -> qpsk, 3 -> bpsk
    config.modulate = 2;
    % 信号Preamble
    config.preamble = [...
        0, 0, 0, 0, 1, 0, 1, 1,...
        0, 1, 1, 1, 0, 1, 1, 1];
end